Compile bug: ngram-mod.h:17:34: error: ‘size_t’ has not been declared

### Git commit

git rev-parse HEAD
971facc38e2544fcf2cc09368de5d1a68e33c10f

i guess it's related to https://github.com/ggml-org/llama.cpp/pull/19164

Adding `#include <cstddef>` to **llama.cpp/common/ngram-mod.h** solve issue.

### Operating systems

Linux

### GGML backends

CUDA

### Problem description & steps to reproduce

Hi there.

Latest build fails on my Debian 12 box with CUDA :

[log.txt](https://github.com/user-attachments/files/24972623/log.txt)

### First Bad Commit
971facc38e2544fcf2cc09368de5d1a68e33c10f

### Compile command

```shell
cmake -B . --fresh -DGGML_CUDA=ON -DCMAKE_CUDA_ARCHITECTURES="86" # -DGGML_CUDA_FA_ALL_QUANTS=ON
cmake --build . --config Release -j$(nproc) --clean-first
```

### Relevant log output

```shell
/whatever/llama.cpp/common/ngram-mod.h:17:34: error: ‘size_t’ has not been declared
...
/whatever/llama.cpp/common/ngram-mod.h:5:1: note: ‘size_t’ is defined in header ‘<cstddef>’; did you forget to ‘#include <cstddef>’?
    4 | #include <vector>
  +++ |+#include <cstddef>
```

--- a/common/ngram-mod.h
+++ b/common/ngram-mod.h
@@ -2,6 +2,7 @@
 
 #include <cstdint>
 #include <vector>
+#include <cstddef>
 
 //
 // common_ngram_mod

